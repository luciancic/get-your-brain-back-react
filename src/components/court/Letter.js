import React, { Component } from 'react';

class Letter extends Component {
    constructor(props) {
        super(props)
        this.audioRef = React.createRef();
        this.audio = require(`../../audio/audio-${this.props.number}.wav`);
    }

    componentDidUpdate() {
        if (this.props.audioShouldPlay) {
            // Calls action creator to mark audioPlayed piece of state as true.
            this.audioRef.current.play();
            this.props.disableAudio();
        }
    }

    render() {
        return <audio src={this.audio} ref={this.audioRef}/>
    }
}

export default Letter;
