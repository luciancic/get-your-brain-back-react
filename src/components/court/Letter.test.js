import React from 'react';
import { shallow } from 'enzyme';
import Letter from './Letter';

it('renders Letter', () => {
  expect( shallow( <Letter number={1}/> )).toMatchSnapshot();
});