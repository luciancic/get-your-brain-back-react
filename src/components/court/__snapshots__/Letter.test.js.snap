// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Letter 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Letter
    number={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "src": "audio-1.wav",
    },
    "ref": Object {
      "current": null,
    },
    "rendered": null,
    "type": "audio",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "audio-1.wav",
      },
      "ref": Object {
        "current": null,
      },
      "rendered": null,
      "type": "audio",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
